File: react.js/src/assets/conversions.js
import {string_chop} from './aux';

export const convertHexToBin = (input) => {
    const binArray = [];
  
    for(let i = 0; i < input.length; i++) {
      binArray.push(parseInt(input[i],16).toString(2).padStart(4, '0'));
    }
  
    return binArray.join("");
  }

  export const convertStringToHex = (input) => {
    const hexArray = [];
  
    for(let i = 0; i < input.length; i++) {
      hexArray.push(input.charCodeAt(i).toString(16));
    }
  
    return hexArray.join("");
}
  
export const convertBinToHex = (input) => {
    const chunkedBinArray = string_chop(input, 4);
    const hexArray = [];

    for (let i = 0; i < chunkedBinArray.length; i++) {
        hexArray.push(parseInt(chunkedBinArray[i], 2).toString(16).toUpperCase())
    }

    return hexArray.join("");
}

export const convertHexToString = (input) => {
    const chunkedHexArray = string_chop(input, 2);
    const strArray = [];

    for (let i = 0; i < chunkedHexArray.length; i++) {
        strArray.push(String.fromCharCode(parseInt(chunkedHexArray[i], 16)));
    }

    return strArray.join("");
}

export const convertStringToBin = (input) => {
    const strASCIIHex = convertStringToHex(input);
    const strASCIIBin = convertHexToBin(strASCIIHex);

    return strASCIIBin;
}

/*
fun convertStringToBin(input: String) : String {
    val stASCIIHex = convertStringToHex(input)
    val stASCIIBin = convertHexToBin(stASCIIHex)

    println("Hex: ${stASCIIHex.chunked(2).joinToString(" ")}")
    println("Binary: ${stASCIIBin.chunked(8).joinToString(" ")}")

    return stASCIIBin
}
*/

File: kotlin/des(cmd)/src/TextConversions/conversions.kt
package TextConversions

import java.math.BigInteger
import org.apache.commons.codec.binary.Base64

fun main() {
    //println(convertHexToString("49206c696b6520636f6d707574657220636c617373"))
   //convertStringToBin("Computer")
    println(convertHexToBase64String("33778ab27cf51a85aaaa9c94551bcfc7754623a960249cc6"))
}

fun convertStringToHex(input: String) : String {
    //val inputNoSpace = input.replace("\\s".toRegex(), "")
    val hexArrayList = arrayListOf<String>()

    input.forEach {
        hexArrayList.add(Integer.toHexString(it.toInt() % 256).padStart(2, '0'))
    }

    return hexArrayList.joinToString("")
}

fun convertBase64StringToHex(input: String) : String {
    //val inputNoSpace = input.replace("\\s".toRegex(), "")
    val hexArrayList = arrayListOf<String>()

    val encodedInputByteArray = input.toByteArray()
    val decodedInputByteArray = Base64.decodeBase64(encodedInputByteArray)

    decodedInputByteArray.forEach {
        var hexInt = it.toInt() % 265
        if (hexInt < 0)
            hexInt += 256

        val hexStr = Integer.toHexString(hexInt).padStart(2, '0')

        hexArrayList.add(hexStr)
    }

    return hexArrayList.joinToString("")
}

fun convertHexToBin(input: String) : String {
    val chunkedHexList = input.chunked(1)
    val binArrayList = arrayListOf<String>()

    chunkedHexList.forEach { hex ->
        binArrayList.add(BigInteger(hex, 16).toString(2).takeLast(4).padStart(4, '0'))
    }

    return binArrayList.joinToString("")
}

fun convertBinToHex(input: String) : String {
    val chunkedBinList = input.chunked(4)
    val hexArrayList = arrayListOf<String>()

    chunkedBinList.forEach { bin->
        hexArrayList.add(BigInteger(bin, 2).toString(16))
    }

    return hexArrayList.joinToString("") { it.toUpperCase() }
}

fun convertHexToString(input: String) : String {
    val chunkedHexList = input.chunked(2)
    val strArrayList = arrayListOf<Char>()

    chunkedHexList.forEach { hex ->
        strArrayList.add(BigInteger(hex, 16).toInt().toChar())
    }

    return strArrayList.joinToString("")
}

fun convertHexToBase64String(input: String) : String {
    val chunkedHexList = input.chunked(2)
    val strArrayList = arrayListOf<Char>()
    val toBeEncodedByteArrayList = arrayListOf<Byte>()

    chunkedHexList.forEach { hex ->
        toBeEncodedByteArrayList.add(BigInteger(hex, 16).toByte())
    }

    val toBeEncodedByteArray = ByteArray(toBeEncodedByteArrayList.size)
    for (i in 0 until toBeEncodedByteArrayList.size) {
        toBeEncodedByteArray[i] = toBeEncodedByteArrayList[i]
    }

    val encodedByteArray = Base64.encodeBase64(toBeEncodedByteArray)

    encodedByteArray.forEach {
        val intVal = it.toInt()
        val charVal = intVal.toChar()
        strArrayList.add(charVal)
    }

    return strArrayList.joinToString("")
}

fun convertStringToBin(input: String) : String {
    val stASCIIHex = convertStringToHex(input)
    val stASCIIBin = convertHexToBin(stASCIIHex)

    println("Hex: ${stASCIIHex.chunked(2).joinToString(" ")}")
    println("Binary: ${stASCIIBin.chunked(8).joinToString(" ")}")

    return stASCIIBin
}

fun convertBase64StringToBin(input: String) : String {
    val stASCIIHex = convertBase64StringToHex(input)
    val stASCIIBin = convertHexToBin(stASCIIHex)

    println("Hex: ${stASCIIHex.chunked(2).joinToString(" ")}")
    println("Binary: ${stASCIIBin.chunked(8).joinToString(" ")}")

    return stASCIIBin
}